### PR TITLE
Tidy up test around tenant attach and restart

### PR DESCRIPTION
**Pull request by hlinnaka**
Clarify the comments to describe what now actually happens, not what used to happen before on-demand download or even earlier.

Remove the redundant checks for 'has_in_progress_downloads'. 'has_in_progress_downloads' is now just an alias for whether the state is Loading or Attaching.

### CODE REVIEW

File: test_runner/regress/test_remote_storage.py [hunk 1]
```diff
@@ -356,7 +344,6 @@ def churn_while_failpoints_active(result):
     def tenant_active():
         all_states = client.tenant_list()
         [tenant] = [t for t in all_states if TenantId(t["id"]) == tenant_id]
-        assert tenant["has_in_progress_downloads"] is False
```
Comment (SomeoneToIgnore) [inline test_runner/regress/test_remote_storage.py:359]: Were we supposed to remove it from the public API too already?

**Review by SomeoneToIgnore — APPROVED** [review]

File: test_runner/regress/test_remote_storage.py (hunk 1 above)
Comment (problame) [reply test_runner/regress/test_remote_storage.py:359]: Console uses it:

https://github.com/neondatabase/cloud/blob/a51b3bef2b32f3af60ad83acb58ad1fc2ba3d78d/goapp/internal/operationsv2/provisioner_common.go#L621-L624

Need to fix that first, but not a blocker.

**Comment by hlinnaka** [comment]
@problame just made similar changes to the comments. I rebased this on top of those changes, this now just rephrases a few comments that I think were still unclear, removes one unnecessary `timeline_detail` call, and the assertions on `has_in_progress_downloads`

**Comment by problame** [comment]
Merged manually.

**Pull request closed without merging**